typescript/src/doc_functionality/lookup.ts

```diff
@@ -147,26 +147,40 @@ export function previousPage(page: DocumentationPage, documentationRoot: Documen
   return null
 }
 
+/** Recursively checks if any visible page is contained inside the structure */ 
+function containsVisiblePage(object: DocumentationPage | DocumentationGroup) {
+  if(object.type === "Group") {
+    return object.children.some((child) => containsVisiblePage(child))
+  }
+
+  return !object.configuration.isHidden
+}
+
 /** Check whether page or group is exportable. Currently page is considered exportable if it doesn't start with underscore */
 export function isExportable(object: DocumentationPage | DocumentationGroup): boolean {
 
   if (object === null || object === undefined) {
     return false
   }
   if (object.type === "Group") {
-    if (object.title.startsWith("_")) {
+    if (
+      object.configuration.isHidden ||
+      object.children.length === 0 ||
+      !containsVisiblePage(object)
```

Comment:
I cannot reuse isExportable since it is also checking parent so it would create recursive loop